include/mbedtls/ssl_internal.h

```diff
@@ -1064,7 +1064,7 @@ int mbedtls_ssl_decrypt_buf( mbedtls_ssl_context const *ssl,
                              mbedtls_record *rec );
 
```

Comment:
How about creating a new header under `library` for these newly shared functions? @mpg @yanesca 

Mbed TLS has no headers under `library`, but Mbed Crypto does and I see no reason why Mbed TLS shouldn't.

Reply:
Yes, I'm all for it!

Reply:
You're suggesting to move the entire `ssl_internal.h` to `library`, correct? Not just the newly added declarations.

Reply:
I'd be happy about moving all `_internal.h` headers to `library`, but I think this should be done in a separate PR. (Also, perhaps after the repo merge will be a bit easier.)